### PR TITLE
Civil servant profile

### LINKED ISSUE #47: Setup Profile View
As a civil servant, I want to have a CRUD for my personal information.
* Authorize that a civil servant can only edit himself.
* Display validation issues using SimpleForm
* Extract Profile view into partials so that they can be rendered also for an organization admin (readonly)

### PR DESCRIPTION
**Pull request by TheFehr**
Closes #47

### CODE REVIEW

File: app/views/civil_servants/civil_servants/_login_information_form.html.slim [hunk 1]
```diff
@@ -0,0 +1,11 @@
+= simple_form_for(civil_servant, url: civil_servants_civil_servant_path, remote: true) do |f|
+  .modal-body
+    = f.simple_fields_for :user do |user_form|
+      = user_form.input :email
+      = user_form.input :language, collection: User.languages.keys,
+              label_method: ->(label) { t(label, scope: %i[activerecord attributes user languages]) },
+              include_blank: false
```
Comment (lukasbischof) [inline app/views/civil_servants/civil_servants/_login_information_form.html.slim:7]: Why is this needed, do we have blank languages?

File: spec/components/modal_component_spec.rb [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe ModalComponent, type: :component do
+  subject(:component) do
+    described_class.new(modal_id: modal_id, title: title)
+  end
+
+  let(:rendered) { render_inline(component) }
```
Comment (lukasbischof) [inline spec/components/modal_component_spec.rb:10]: I'm not sure if the component really should be the subject since you only test against `rendered`

File: spec/components/tabular_card_component_spec.rb [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  context 'with the body content_area' do
+    subject(:component) do
+      described_class.new(title: title, table_content: table_content)
+    end
+
+    let(:rendered) do
+      render_inline(component) do |component|
+        component.with(:actions, action_content)
+      end
+    end
```
Comment (lukasbischof) [inline spec/components/tabular_card_component_spec.rb:29]: Also here not really sure about the let/subject

File: spec/components/tabular_card_component_spec.rb [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  end
+
+  let(:rendered) { render_inline(component) }
+  let(:title) { 'My title' }
+  let(:table_content) { { Test: 'Hans' } }
+
+  it 'renders id and title' do
+    expect(rendered.css('h4').text).to eq title
+    expect(rendered.css('.mb-lg-1.col-12.col-xxl-5.font-weight-bold').text).to eq table_content.keys.first.to_s
+    expect(rendered.css('.mb-1.col-12.col-xxl-7').text).to eq table_content.values.first.to_s
```
Comment (lukasbischof) [inline spec/components/tabular_card_component_spec.rb:17]: I'd prefer to not use css classes for testing since it's only purpose should be styling. If we decide to have a greater margin, then tests would suddenly fail. I'd suggest to use data attributes (like I did in the cards component, e.g. `.mb-lg-1.col-12.col-xxl-5.font-weight-bold[data-content]` and then use
```suggestion
    expect(rendered.css('[data-content]').text).to eq table_content.keys.first.to_s
    expect(rendered.css('.mb-1.col-12.col-xxl-7').text).to eq table_content.values.first.to_s
```
in specs and/or JavaScript)

File: spec/requests/civil_servants/civil_servants_controller_spec.rb [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    let(:request_params) do
+      {
+        civil_servant: update_params,
+        form_partial: 'personal_information_form'
+      }
+    end
+    let(:update_params) do
+      {
+        first_name: 'Lara',
+        last_name: 'Croft',
```
Comment (lukasbischof) [inline spec/requests/civil_servants/civil_servants_controller_spec.rb:35]: Lol, this spec data seems quite familiar to me…

File: spec/services/expense_sheet_generator_spec.rb [hunk 6]
```diff
@@ -20,7 +20,7 @@
       end
     end
 
-    it 'sets correct civil_servant', :aggregate_failures do
+    it 'sets correct civil_servants', :aggregate_failures do
```
Comment (lukasbischof) [inline spec/services/expense_sheet_generator_spec.rb:23]: Aggregation is now default, you can remove it

```suggestion
    it 'sets correct civil_servants' do
```

File: spec/views/civil_servants/civil_servants/edit.html.slim_spec.rb [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+RSpec.describe 'civil_servants/civil_servants/edit.html.erb', type: :view do
+  before do
+    sign_in user
+    assign(:civil_servant, civil_servant)
+
+    render
+  end
+
+  let(:civil_servant) { create(:civil_servant, :full) }
```
Comment (lukasbischof) [inline spec/views/civil_servants/civil_servants/edit.html.slim_spec.rb:13]: You do not need to `create` when you're not doing queries in the view. This would improve spec performance

File: spec/views/civil_servants/civil_servants/edit.html.slim_spec.rb [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+      t('civil_servants.civil_servants.edit.cards.login_information.title'),
+      t('civil_servants.civil_servants.edit.cards.bank_and_insurance_information.title'),
+      t('civil_servants.civil_servants.edit.cards.service_specific_information.title')
+    ]
+  end
+
+  it 'displays all the information' do
+    expected_strings.each do |value|
+      expect(rendered).to include(value.to_s)
+    end
```
Comment (lukasbischof) [inline spec/views/civil_servants/civil_servants/edit.html.slim_spec.rb:49]: Maybe also something like this

```suggestion
    expect(rendered).to include(*expected_string.map(&:to_s))
```

File: app/services/pdfs/service_agreement/german_form_fields.rb [hunk 9]
```diff
@@ -42,12 +42,12 @@ module GermanFormFields
...
+        # 'Ort Datum' => ->(service:) { service.civil_servants.signature_location_and_date },
         # 'Ort Datum_2' => ->(service:) { service.service_specification.organization.signature_location_and_date },
-        # 'D' => ->(service:) { service.civil_servant.workshop_language_german? },
-        # 'E' => ->(service:) { service.civil_servant.workshop_language_english? },
-        # 'F' => ->(service:) { service.civil_servant.workshop_language_french? },
-        # 'I' => ->(service:) { service.civil_servant.workshop_language_italian? },
+        # 'D' => ->(service:) { service.civil_servants.workshop_language_german? },
+        # 'E' => ->(service:) { service.civil_servants.workshop_language_english? },
+        # 'F' => ->(service:) { service.civil_servants.workshop_language_french? },
+        # 'I' => ->(service:) { service.civil_servants.workshop_language_italian? },
```
[the remark this answers is not in the document]
Comment (lukasbischof) [reply app/services/pdfs/service_agreement/german_form_fields.rb:50]: Why does a service now have multiple civil_servants?

File: spec/components/tabular_card_component_spec.rb (hunk 4 above)
Comment (lukasbischof) [reply spec/components/tabular_card_component_spec.rb:17]: See also https://github.com/my-zivi/my-zivi/blob/develop/spec/components/overview_card_component_spec.rb#L20 or https://github.com/my-zivi/my-zivi/blob/develop/spec/components/overview_card_component_spec.rb#L33

File: app/services/pdfs/service_agreement/german_form_fields.rb (hunk 9 above)
[the remark this answers is not in the document]
Comment (TheFehr) [reply app/services/pdfs/service_agreement/german_form_fields.rb:50]: This is a broken refactoring. Luckily only in the comments

File: app/views/civil_servants/civil_servants/_login_information_form.html.slim (hunk 1 above)
Comment (TheFehr) [reply app/views/civil_servants/civil_servants/_login_information_form.html.slim:7]: No but the input has by default a blank option. Which we don't want since it would be invalid

File: spec/components/tabular_card_component_spec.rb (hunk 4 above)
Comment (TheFehr) [reply spec/components/tabular_card_component_spec.rb:17]: I wasn't very happy either that sounds smart. Why not an id?

File: spec/requests/civil_servants/civil_servants_controller_spec.rb (hunk 5 above)
Comment (TheFehr) [reply spec/requests/civil_servants/civil_servants_controller_spec.rb:35]: Hm I don't know why that could be

File: spec/views/civil_servants/civil_servants/edit.html.slim_spec.rb (hunk 7 above)
Comment (TheFehr) [reply spec/views/civil_servants/civil_servants/edit.html.slim_spec.rb:13]: I think it didn't work with build. But I will recheck for you.

File: app/services/pdfs/service_agreement/german_form_fields.rb (hunk 9 above)
[the remark this answers is not in the document]
Comment (lukasbischof) [reply app/services/pdfs/service_agreement/german_form_fields.rb:50]: Yes I already thought that. Thanks, RubyMine 😂

File: app/views/civil_servants/civil_servants/_login_information_form.html.slim (hunk 1 above)
Comment (lukasbischof) [reply app/views/civil_servants/civil_servants/_login_information_form.html.slim:7]: Ah, makes sense

Comment (lukasbischof) [reply app/views/civil_servants/civil_servants/_login_information_form.html.slim:7]: Even if you specify `required: true`

File: spec/components/tabular_card_component_spec.rb (hunk 4 above)
Comment (lukasbischof) [reply spec/components/tabular_card_component_spec.rb:17]: Because you render your card component multiple times and id's are unique (It's actually not valid XHTML. Also a selector would be ambiguous). data attributes can appear multiple times as you please

File: app/views/civil_servants/civil_servants/_login_information_form.html.slim (hunk 1 above)
Comment (TheFehr) [reply app/views/civil_servants/civil_servants/_login_information_form.html.slim:7]: Yes

File: spec/components/tabular_card_component_spec.rb (hunk 4 above)
Comment (lukasbischof) [reply spec/components/tabular_card_component_spec.rb:14]: image

Comment (TheFehr) [reply spec/components/tabular_card_component_spec.rb:14]: True true

File: spec/views/civil_servants/civil_servants/edit.html.slim_spec.rb (hunk 7 above)
Comment (TheFehr) [reply spec/views/civil_servants/civil_servants/edit.html.slim_spec.rb:13]: `ActionView::Template::Error: undefined method 'zdp' for nil:NilClass`

File: spec/components/modal_component_spec.rb [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+
+  context 'with body content_area' do
+    let(:rendered) do
+      render_inline(component) do |component|
+        component.with(:body, body_content)
+      end
+    end
+    let(:component) do
+      described_class.new(modal_id: modal_id, title: title) do |component|
+        component.with(:body, 'Hello from the body')
```
Comment (lukasbischof) [inline spec/components/modal_component_spec.rb:25]: Why two times component.with?
